.github/automation/commit-msg-check.py

```diff
@@ -20,29 +20,17 @@
 
 import argparse
 import subprocess
+import re
 
 # * Ensuring the scopes end in colon and same level scopes are comma delimited.
 # TODO: Limit scopes to an acceptable list of tags.
 def __scopeCheck(msg: str):
     status = "Message scope: "
 
-    firstLine = (msg.partition("\n")[0]).strip()
-
-    if not ":" in firstLine:
-        print(f"{status} FAILED: Commit message does not include scope")
+    if not re.match('^[a-z0-9]+: ', msg):
+        print(f"{status} FAILED: Commit message must follow the format <scope>:[ <scope>:] <short description>")
         return False
 
-    # The last element of the split is the title, which we don't care about.
-    scopesArray = firstLine.split(":")[:-1]
-
-    for scopes in scopesArray:
-        numWords = len(scopes.split())
-        numCommas = scopes.count(",")
-
-        if numWords != numCommas + 1:
-            print(f"{status} FAILED: Same-level scopes must be comma-separated. Bad token: '{scopes}'")
-            return False
```

Comment:
I don't see any issue with keeping this check for the top level scope. At minimum, the comment on line 25 should be updated if this is removed.

Reply:
Right. Technically coding guidelines do not allow comma for top level scope, though I see that we do that from time to time in the commit history.